packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -505,3 +504,9 @@ export interface SdkPackage<TServiceOperation extends SdkServiceOperation> {
 export type SdkHttpPackage = SdkPackage<SdkHttpOperation>;
 
 export type LanguageScopes = "dotnet" | "java" | "python" | "javascript" | "go" | string;
+
+export enum UsageFlags {
```

Comment:
Can we name this ModelUsageFlags so it doesn't get confused with other kinds of usage?

Reply:
or `ClientUsage`

Reply:
after verification, this will break emitter authors who are currently filtering using `@typespec/compiler`'s `UsageFlags`. Going to improve docs instead

Reply:
it doesn't break if it is only used with bitwise operations https://www.typescriptlang.org/play?#code/KYOwrgtgBAYg9nKBvAUFdUCWIAOYAuUAvFAIxQA8FZANGhnAXoSeVVAEx0C+KKokKADlgAd3iJUGLLgLEylaqTrTG+ZvLbUuKXigBmYEAGN8mOCCgBzYPgkAKAJQAuEeITJ66AE62w3yzcJADpsZgBuXT5ogHoYmSgAG0wAIz5jCwBnQn0PEhs7BCdIlEx9e1zEADJYBFDZfEdPXSA

if however you used the type and pass it around it will fail